packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -472,14 +454,34 @@ export class DocumentDeltaConnection
 
 		this._details = await new Promise<IConnected>((resolve, reject) => {
 			const failAndCloseSocket = (err: IAnyDriverError) => {
-				this.closeSocket(err);
+				try {
+					this.closeSocket(err);
+				} catch (failError) {
+					const normalizedError = this.addPropsToError(failError);
+					this.logger.sendErrorEvent({ eventName: "CloseSocketError" }, normalizedError);
+				}
 				reject(err);
 			};
 
 			const failConnection = (err: IAnyDriverError) => {
-				this.disconnect(err);
+				try {
+					this.disconnect(err);
```

Comment:
I'd hope that this is not required, as Container (and probably DM) will use EventEmitter that captures exceptions in handlers. Or do we think that there is an excetpion in our code processing disconnect?

Reply:
assert 0x416 hits during closeSocket. So, I want to log that and other state when that happens. Other PR: https://github.com/microsoft/FluidFramework/pull/14021 should prevent 0x416 from happening but I want to add more telemetry as it is affecting a good number of loop users.